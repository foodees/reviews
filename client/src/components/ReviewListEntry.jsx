
import React from 'react';
import ReactStars from 'react-stars';
import moment from 'moment';

var ReviewListEntry = (props) => {

 if (props) {
   var text = props.review.text.replace(/\\\\/g, /\\/).replace(/\\'/g, '\'').replace(/\\"/g, '\"');
   var textArr = text.split('\\n');
   var textWithSpacing = textArr.map((paragraph, idx) =>
     <p key={idx}>{paragraph}</p>
   );
 }
 var rating = props.review.stars;
 var starPosition = -420;
 if (rating >= 1) {
   starPosition += ((rating * 2 * -14) + 14);
 }
 var percentage = props.rating/5 * 100;

 return(
   <div className="review-list-entry">
     <div className="user-contents">
       <div><img src="https://s3-us-west-1.amazonaws.com/foodee-reviews/userpic.png" alt="user image"></img></div>
       <div>
         <div style={{color: 'blue', fontWeight: 'bold'}}>{props.user.name}</div><br></br>
         <div><span><i className="fas fa-users fa-lg" style={{color: 'Tomato'}}></i> {props.user.fans} friends</span></div><br></br>
         <div><span><i className="far fa-star fa-lg" style={{color: 'Tomato'}}></i> {props.user.review_count} reviews</span></div>
       </div>
     </div>
     <div className="review-contents">
       <div className="review-rating"><span style={{'backgroundPositionY':starPosition + 'px'}} className="nearMeListStars"></span> {moment(props.review.date, moment.ISO_8601).format('l')}</div>
       <span className="review-text">{textWithSpacing}</span>
       <div className="review-compliments"> Useful: {props.review.useful} | Funny: {props.review.funny} | Cool: {props.review.cool}</div>
     </div>
   </div>
 );

}

export default ReviewListEntry;
