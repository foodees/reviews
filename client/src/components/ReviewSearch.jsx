import React from 'react';

var ReviewSearch = (props) => (
  <div id="reviews-search">
    <span id="reviews-search-bar">
      <input type="text" onChange={props.setSearchTerm}/>
      <button onClick={props.searchReviews}> Search Comments </button>
    </span>
    <span id="reviews-sort"> Sort By:
      <select id="sortBy" onChange={props.sort}>
        <option value="newestFirst" defaultValue>Newest First</option>
        <option value="oldestFirst">Oldest First</option>
        <option value="highestRated">Highest Rated</option>
        <option value="lowestRated">Lowest Rated</option>
      </select>
    </span>
  </div>
);

//"^4.5.0"
//"^2.2.1"

export default ReviewSearch;
