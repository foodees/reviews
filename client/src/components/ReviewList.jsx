
import React from 'react';
import ReviewListEntry from './ReviewListEntry.jsx';

var ReviewList = (props) => (
  <div className="review-list">
    {props.reviews.map((review, index) => {
      for (var i = 0 ; i < props.users.length; i++) {
        if (props.users[i].id === props.reviews[index].user_id) {
          return <ReviewListEntry review={review} user= {props.users[i]} key={index}/>
        }
      }
    })}
  </div>
);

export default ReviewList;
