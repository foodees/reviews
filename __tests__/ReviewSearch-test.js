import React from 'react';
// const Enzyme = require('enzyme');
// const EnzymeAdapter = require('enzyme-adapter-react-16');
import { shallow, mount, render } from 'enzyme';
import renderer from 'react-test-renderer';

// import Adapter from 'enzyme-adapter-react-15';
var userData = require('../__mocks__/userData.js');
var reviewData = require('../__mocks__/reviewData.js');

import App from '../client/src/app.jsx';
import ReviewList from '../client/src/components/ReviewList.jsx';
import ReviewListEntry from '../client/src/components/ReviewListEntry.jsx';
import ReviewSearch from '../client/src/components/ReviewSearch.jsx';

describe('example component', function() {

  it('should contain a search bar and sorting drop down menu', function() {
    let wrapper = shallow(<div className="some-class other-class" />);
    let bool = wrapper.is('.some-class');
    expect(wrapper.is('.some-class')).toBe(bool);
  });
});

describe('App component', () => {
  it('should render without throwing an error', () => {
    var app = mount(<App />);
    // console.log('app: ', app);
    expect(app.find('#reviews')).toHaveLength(1);
    
    // var app2 = shallow(<App />);
    app.find('#sortBy').simulate('change', {target: {value: 'oldestFirst'}});
    console.log('app sortBy state: ', app);
    expect(app.state('sortBy')).toEqual('oldestFirst');
    
    var app2 = renderer.create(<App />).toJSON();
    expect(app2).toMatchSnapshot();
    // app.find('#sortBy').simulate('change', {target: {value: 'oldestFirst'}});
    // expect(app.state('sortBy')).toEqual('oldestFirst');
    // expect(app.contains('#review-list')).toBe(true);
    // expect(app.contains('#review-list-entry')).toBe(true);
  });
});

describe('ReviewList component', () => {
  it('should render without throwing an error', () => {
    // console.log('USER DATA: ', userData.pageOne);
    // console.log('REVIEW DATA: ', reviewData.pageOne);
    var list = shallow(<ReviewList reviews={reviewData.pageOne} users={userData.pageOne} />);
    expect(list.is('.review-list')).toBe(true);
    expect(list.find(ReviewListEntry)).toHaveLength(10);
  });
});

describe('ReviewListEntry component', () => {
  it('should render without throwing an error', () => {
    var listEntry = shallow(<ReviewListEntry user={userData.pageOne[0]} review={reviewData.pageOne[0]} />);
    expect(listEntry.is('.review-list-entry')).toBe(true);
    expect(listEntry.find('.user-contents')).toHaveLength(1);
    expect(listEntry.find('.review-contents')).toHaveLength(1);
  });
});

describe('ReviewSearch component', () => {
  it('should render without throwing an error', () => {
    var search = shallow(<ReviewSearch setSearchTerm="steak" searchReviews={reviewData.pageOne} sort={() => { console.log('sort');}}/>);
    expect(search.is('#reviews-search')).toBe(true);
  });
});