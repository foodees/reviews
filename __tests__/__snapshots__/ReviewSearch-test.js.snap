// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render without throwing an error 1`] = `
<div
  id="reviews"
>
  <h2>
    <span
      id="review-header"
    >
      Recommended Reviews
    </span>
     for 
    
  </h2>
  <div
    id="reviews-search"
  >
    <span
      id="reviews-search-bar"
    >
      <input
        id="reviews-input"
        onChange={[Function]}
        type="text"
      />
      <button
        id="reviews-button"
        onClick={[Function]}
      />
    </span>
    <span
      id="reviews-sort"
    >
       
      <span>
        Sort By: 
      </span>
      <select
        id="sortBy"
        onChange={[Function]}
      >
        <option
          defaultValue={true}
          value="newestFirst"
        >
          Newest First
        </option>
        <option
          value="oldestFirst"
        >
          Oldest First
        </option>
        <option
          value="highestRated"
        >
          Highest Rated
        </option>
        <option
          value="lowestRated"
        >
          Lowest Rated
        </option>
      </select>
    </span>
  </div>
  <div
    className="review-list"
  />
  <ul
    className={undefined}
  >
    <li
      className="pages disabled"
    >
      <a
        className={undefined}
        href={undefined}
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex="0"
      >
        &lt; Prev
      </a>
    </li>
    <li
      className="pages selectedPage"
    >
      <a
        aria-current="page"
        aria-label="Page 1 is your current page"
        className={undefined}
        href={undefined}
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex="0"
      >
        1
      </a>
    </li>
    <li
      className="pages disabled"
    >
      <a
        className={undefined}
        href={undefined}
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex="0"
      >
        Next &gt;
      </a>
    </li>
  </ul>
</div>
`;
